alea/submitter.py

```diff
@@ -0,0 +1,498 @@
+        loglevel (str, optional (default='INFO')): the log level
+
+    Keyword Args:
+        kwargs: the arguments of __init__ method of the Submitter,
+            containing configurations of clusters
+
+    Caution:
+        All the source of template should be from the same folder.
+        All the output, including toydata and fitting results, should be in the same folder.
+
+    """
+
+    config_file_path: str
+    template_path: str
+    allowed_special_args: List[str] = []
+    logging = logging.getLogger("submitter_logger")
+
+    def __init__(
+        self,
+        statistical_model: str,
+        statistical_model_config: str,
+        poi: str,
+        computation_options: dict,
+        computation: str = "discovery_power",
+        outputfolder: Optional[str] = None,
+        debug: bool = False,
+        loglevel: str = "INFO",
+        **kwargs,
+    ):
+        """Initializes the submitter."""
+        if type(self) == Submitter:  # noqa: E721
+            raise RuntimeError(
+                "You cannot instantiate the Submitter class directly, "
+                "you must use a subclass where the submit method are implemented"
+            )
+        loglevel = getattr(logging, loglevel.upper())
+        self.logging.setLevel(loglevel)
+
+        # find the path of template, requires users install alea-inference properly
+        self.run_toymc = shutil.which("alea-run_toymc")
+        if self.run_toymc is None:
+            raise RuntimeError(
+                "alea-run_toymc is not found, "
+                "please make sure you have installed alea-inference correctly."
+            )
+
+        self.statistical_model = statistical_model
+        self.statistical_model_config = statistical_model_config
+        self.poi = poi
+        self.outputfolder = outputfolder
+
+        self.computation = computation_options[computation]
+        self.debug = debug
+
+        # Find statistical model config file
+        if not os.path.exists(self.statistical_model_config):
+            self.statistical_model_config = os.path.join(
+                os.path.dirname(get_file_path(self.config_file_path)), self.statistical_model_config
+            )
+        if not (
+            os.path.exists(self.statistical_model_config)
+            and os.path.isfile(self.statistical_model_config)
+        ):
+            raise FileNotFoundError(
+                f"statistical_model_config {self.statistical_model_config} "
+                "is not a valid filename or does not exist, "
+                "presumably it should be in the same folder as "
+                f"config_file_path {self.config_file_path}."
+            )
+
+        # Initialize the statistical model
+        statistical_model_class = StatisticalModel.get_model_from_name(self.statistical_model)
+        self.model = statistical_model_class.from_config(
+            self.statistical_model_config, template_path=self.template_path
+        )
```

Comment:
I think `Submitter` now is only responsible for `BlueiceExtendedModel`.

Reply:
But shouldn't we keep it as general as possible? At least, people should know which requirements they should fulfil in case they want to just write their own model file and use the rest of our framework. For me this is an important selling point for alea or do you think this is not realistic?

Reply:
I think in general the submitter is configuration-file-based. There are three cases in my mind:

1. If the model is simple, like `GaussianModel`, then the user does not need a submitter.
2. If the model is complex: 
2.1.  If the user wants to use `BlueiceExtendedModel`, then it is this PR.
2.2. If the user wants another class of model, then they can just inherit `SubmitterSlurm` or other submitter to define a new submitter, and overwrite `computation_tickets_generator`. The user is still able to use `utilix.batchq` related submission script.

Reply:
So the generalization ability of `Submitter.__init__` and `Submitter.computation_tickets_generator` is not so high but users can still overwrite them.

Reply:
That sounds good at least for now I'd say. In case other people add something we might at some point consider making this more general but right now there is no need